vividus-plugin-web-app/src/main/java/org/vividus/crawler/CrawlControllerFactory.java

```diff
@@ -69,21 +78,15 @@ public CrawlController createCrawlController(URI mainApplicationPage)
 
     private CrawlConfig createCrawlConfig(URI mainApplicationPage) throws IOException
     {
-        CrawlConfig crawlConfig = new CrawlConfig();
+        CrawlConfig crawlConfig = propertyMapper.readValue(defaultCrawlConfig, MappedCrawlConfig.class);
```

Comment:
Why not tell the mapper to ignore the unknown fields instead? 
something like
`javaPropsMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);` ?

Reply:
we share one javaPropsMapper in PropertyMapper for all mapping operations, if we want to use this we will have to perform clone of the javaPropsMapper and then apply the FAIL_ON_UNKNOWN_PROPERTIES config, so for me current way of things is more lightweight

Reply:
I see no ~heaviness~ problems having a "configurable" PM  in the same flavour as how it's done for HTTP clients. I believe there are multiple ways to achieve this. New factory method, configurable mapper, etc

Reply:
I don't see any advantage of creating the whole property mapper for one small case, additionally by having the annotation we see that serialisation does not fail on unknown property whiich improves understanding of the code whereas no one looks into xml configs, so please provide any advantage, I do not see any issues in using annotation

Reply:
@valfirst  Is this link advocating for this kostyl? :)

Reply:
@valfirst @ikalinin1 another suggestion is to reuse the readMapAs logic on our side and use ObjectMapper methods that allows to define DeserializationFeature for current deserialisation and that won't apply in other places